akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala

```diff
@@ -363,17 +367,25 @@ private final class RestartWithBackoffFlow[In, Out](
   val out = Outlet[Out]("RestartWithBackoffFlow.out")
 
   override def shape = FlowShape(in, out)
+
   override def createLogic(inheritedAttributes: Attributes) = new RestartWithBackoffLogic(
     "Flow", shape, minBackoff, maxBackoff, randomFactor, onlyOnFailures, maxRestarts) {
+    val delay = inheritedAttributes.get[Delay](Delay(50.millis)).duration
 
     var activeOutIn: Option[(SubSourceOutlet[In], SubSinkInlet[Out])] = None
 
     override protected def logSource = self.getClass
 
     override protected def startGraph() = {
-      val sourceOut = createSubOutlet(in)
-      val sinkIn = createSubInlet(out)
-      Source.fromGraph(sourceOut.source).via(flowFactory()).runWith(sinkIn.sink)(subFusingMaterializer)
+      val sourceOut: SubSourceOutlet[In] = createSubOutlet(in)
+      val sinkIn: SubSinkInlet[Out] = createSubInlet(out)
+
+      Source.fromGraph(sourceOut.source)
+        // Temp fix while waiting cause of cancellation. See #23909
+        .via(RestartWithBackoffFlow.delayCancellation[In](delay))
+        .via(flowFactory())
+        .runWith(sinkIn.sink)(subFusingMaterializer)
```

Comment:
So here, the attributes will not be passed on from the `RestartWithBackoffLogic` stage, unless there is something I don't know about the `subFusingMaterializer`.

I think you'd expect that `restart(inner flow).withAttributes(Greatness(max))` would provide that attribute as a default for the inner flow when materialized as well. Could be that is missing currently.

Reply:
Yes i'd expect that, you can put them on the inner flow but not on the restart bit for the inner flow